### DISCUSSION TITLE
Taiko classic mod does not match stable in various areas

### DISCUSSION THREAD

**Discussion by vunyunt** [ppy/osu#19580] [Ideas] [answered]
Related to and expands on https://github.com/ppy/osu/discussions/17265 and https://github.com/ppy/osu/issues/14794

---

These issues are discovered while I was working on taiko diffcalc. There are a few issues with taiko classic that does not allow for a proper baseline for reading/sv consideration, since players can manipulate hidden time and note screen-time by resizing the window. The list of issues I've found are:
- NM does not limit the aspect ratio by fading the notes in
- HD is not locked to a 4:3 aspect ratio
- HR is not locked to a 2:1-ish aspect ratio for wide display aspect ratios

I've made a quick-ish WIP implementation at my repository here: https://github.com/vunyunt/osu/tree/taiko-classic-fix (this is not a PR because the implementation is not ready yet), which is included in the comparison screenshots below (lazer screenshots are with the classic mod enabled):

<details>
<summary>NM</summary>
<img src="https://user-images.githubusercontent.com/5846322/182918130-b2fa2e7a-d638-4d67-92f9-a148d83b4169.png"/>
</details>
<details>
<summary>HD</summary>
Seems like I have unnecessarily changed the time range, but I'll need to look further into it
<img src="https://user-images.githubusercontent.com/5846322/182918171-42a42034-08fb-4758-92b3-84bd6956de11.png"/>
</details>
<details>
<summary>HR</summary>
<img src="https://user-images.githubusercontent.com/5846322/182918278-98cd09ab-c370-4628-8ecb-4f6b3180794d.png"/>
</details>
<details>
<summary>HDHR</summary>
<img src="https://user-images.githubusercontent.com/5846322/182918307-ef06ce3a-4132-4cb6-962f-00e7d54ce098.png"/>
</details>

---

These are the assumptions I've made based on eyeballing stable, thus they are probably not super accurate:
 - For NM, Stable limits the aspect ratio to around 2:1 by fading the notes in.
 - For HD, Stable locks the aspect ratio to 4:3 by cutting the play field off.
 - For HD, Stable seems to also slow the SV down somewhat, but judging from the screenshot that might not be true.
 - For HR, Stable limits the aspect ratio to around 2:1 by stretching the time range.
 - For HDHR, HR changes to the playfield override HD changes.

Before further adjustments/working further on it, I would like to confirm the assumptions I've made or if there's anything I've missed, and that if I'm allowed to work on it at all. Thank you for reading in advance!

**Comment by peppy** [comment, marked as answer]
I feel like these taiko nuances are never going to end. If anything, we really need to simplify this going forward.

I think as long as your changes are aiming to match stable, then you're more than welcome to PR them. Please split up any changes if they are large, otherwise bundling all the mentioned changes together may be okay.

**Reply by vunyunt** [reply]
Agreed on the simplification part, personally I really like lazer's non-classic approach where the aspect ratio and time range is just locked, thereby providing an identical play field to all display aspect ratios and mod combinations.

The original motivation for this was so that we can have a baseline for diffcalc, but they are aimed toward matching stable. Changes shouldn't be too large code wise, and most of it is handled within `TaikoModClassic`.
